### PR TITLE
Feature/merge localized with technologies

### PR DESCRIPTION
**Pull request by MargaretaKusan**
This pull request contains merged branches: ContactForm, LocalizedStrings & mergeTechnologiesCore. Instead of merging the two currently open pull requests (ContactForm and mergeTechnologiesCore) we can just merge this one into Develop and everything is rainbows and unicorns.

### CODE REVIEW

File: HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormInteractor.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import BSWFoundation
+import HiringAppCore
+
+class ContactFormInteractor {
+
+    //MARK: - Stored properties
+    let repository: ContactFormRepositoryProtocol
+    
+    //MARK: - Initializer
+    init?(repository: ContactFormRepositoryProtocol? = ContactFormRepository()) {
```
Comment (Alegege) [inline HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormInteractor.swift]: Why is it allowed to repository parameter to be optional?

File: HiringApp/HiringApp/Application/Extensions/UIButton.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+//  HiringApp
+//
+//  Created by Margareta Kusan on 03/07/2017.
+//  Copyright © 2017 Visual Engineering. All rights reserved.
+//
+
+import UIKit
+
+extension UIButton {
+    override open var isHighlighted: Bool {
```
Comment (daiku60) [inline HiringApp/HiringApp/Application/Extensions/UIButton.swift]: wrong!!!!
you shall **never** __ever__ override in an extension!

File: HiringApp/HiringApp/Application/Extensions/UIButton.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        didSet {
+            if isHighlighted {
+                self.backgroundColor = self.backgroundColor?.withAlphaComponent(0.5)
+            } else {
+                self.backgroundColor = self.backgroundColor?.withAlphaComponent(1)
+            }
+        }
+    }
+    
+    override open var isEnabled: Bool {
```
Comment (daiku60) [inline HiringApp/HiringApp/Application/Extensions/UIButton.swift]: same as above

File: HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormPresenter.swift [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        self.view = view
+    }
+    
+    //MARK: - Private API
+    fileprivate func sendContactFormData(candidate: ContactFormViewModel) {
+        interactor.sendContactFormData(candidate: candidate).upon(.main) { result in
+            switch result {
+            case .failure(let error):
+                self.state = .error(error)
+                showAlert(baseView: self.view as! UIViewController, title: "Error", message: "Error occured, please try again.")
```
Comment (jordiaguila) [inline HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormPresenter.swift]: The showAlert method should call a method in the userinterface protocol in order to transfer alert presentation responsability to a UIKit dependant element.

Check also using force unwrap

File: HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormPresenter.swift [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+extension ContactFormPresenter: ContactFormPresenterProtocol {
+
+    func viewDidLoad() {
+        self.viewModel = ContactFormViewModel()
+    }
+    
+    func tappedSendButton() {
+        guard let unwrappedViewModel = viewModel, self.viewModel?.validate() == true else {
```
Comment (Alegege) [inline HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormPresenter.swift]: You can use the same variable name in cases like this one: 

`guard let viewModel = viewModel`

File: HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormViewController.swift [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+    static let brandBlue: UIColor =  UIColor(red: 101/255.0, green: 174/255.0, blue: 242/255.0, alpha: 1.0)
+    static let highlightedButtonColor: UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 0.5)
+    static let titleLabelTextColor: UIColor = .white
+    static let stackViewBackgroundColor: UIColor = .white
+    static let formLabelTextColor: UIColor = .black
+    static let wrongInputColor: UIColor = .red
+    static let correctInputColor: UIColor = .darkGray
+    
+    // Text
+    static let titleLabelText = R.string.localizable.cf_title()
```
Comment (jordiaguila) [inline HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormViewController.swift]: Why saving text resources in static properties? What happens when an application has a inner method to update its   displayed language?

File: HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormViewController.swift [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+
+        layout()
+        presenter.viewDidLoad()
+        
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.view.layoutIfNeeded()
+        sendButton.layer.cornerRadius = sendButton.frame.size.height * Constants.sendButtonCornerRadiusMultiplier
```
Comment (jordiaguila) [inline HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormViewController.swift]: Forcing view layout (self.view.layoutIfNeeded()) to allow proper CornerRadius computation is a bad idea. The cornerRadius should be applied by the button itself, maybe a UIButton subclass?

File: HiringApp/HiringApp/Application/Scenes/ContactFormSent/ContactFormSent.swift [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+private enum Constants {
+    // Colors
+    static let brandBlue: UIColor =  UIColor(red: 101/255.0, green: 174/255.0, blue: 242/255.0, alpha: 1.0)
+    static let highlightedButtonColor: UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 0.5)
+    static let buttonTextColor: UIColor = .white
+    static let labelBackgroundColor: UIColor = .clear
+    static let buttonBackgroundColor: UIColor = .white
+    
+    // Text
+    static let labelText = R.string.localizable.cf_formSentTitle()
```
Comment (jordiaguila) [inline HiringApp/HiringApp/Application/Scenes/ContactFormSent/ContactFormSent.swift]: Store text resources in static constants

File: HiringApp/HiringAppCore/Repositories/ContactFormRepository.swift [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    func sendContactFormData(candidate: CandidateModel) -> Task<()>
+}
+
+public class ContactFormRepository {
+    
+    //MARK: - Stored properties
+    let apiProvider: APIProviderType
+    let deviceID: String? = UIDevice.current.identifierForVendor?.uuidString
+    
+    public init?(apiProvider: APIProviderType = APIProvider()) {
```
Comment (jordiaguila) [inline HiringApp/HiringAppCore/Repositories/ContactFormRepository.swift]: Why failable init?

**Pull request merged**